monetdbe/_cffi/internal.py

```diff
@@ -275,24 +307,44 @@ def dump_table(self, schema_name: str, table_name: str, backupfile: Path):
 
     def get_columns(self, table: str, schema: str = 'sys') -> Iterator[Tuple[str, int]]:
         self._switch()
-        count_p = ffi.new('size_t *')
-        names_p = ffi.new('char ***')
-        types_p = ffi.new('int **')
+        count_p = ffi.new('size_t*')
+        columns_p = ffi.new('monetdbe_column**')
 
-        lib.monetdbe_get_columns(self._monetdbe_database, schema.encode(), table.encode(), count_p, names_p, types_p)
+        lib.monetdbe_get_columns(self._monetdbe_database, schema.encode(), table.encode(), count_p, columns_p)
 
         for i in range(count_p[0]):
-            name = ffi.string(names_p[0][i]).decode()
-            type_ = types_p[0][i]
+            name = ffi.string(columns_p[0][i].name).decode()
+            type_ = columns_p[0][i].type
             yield name, type_
 
 
 from monetdbe._cffi.branch import newer_then_jul2021
 if not newer_then_jul2021:
+    def bind(statement: monetdbe_statement, data: Any, parameter_nr: int, type_info=None) -> None:
+        prepared = prepare_bind(data)
+        check_error(lib.monetdbe_bind(statement, prepared, parameter_nr))
+
     def prepare(self, query: str) -> monetdbe_statement:
         self._switch()
         stmt = ffi.new("monetdbe_statement **")
         check_error(lib.monetdbe_prepare(self._monetdbe_database, str(query).encode(), stmt))
-        return stmt[0]
+        return stmt[0],
```

Comment:
what is this comma here

Reply:
There are two signatures for prepare: the old one and the new one. Having prepare always return a tuple keeps the signature of both methods the same. Which makes it easier to minimize the effect of the API changes in monetdbe-python. See the definition of `_execute_monetdbe`.
https://github.com/MonetDBSolutions/MonetDBe-Python/blob/440d25fbda9fbbde72f7e920fb17d35f0e0bd0f0/monetdbe/cursors.py#L126-L137

Reply:
Will have a cleaner solutions once Jul2021 has become obsolete.

Reply:
A cleaner solution becomes immediate once Jul2021 has become obsolete.